### PR TITLE
[Refactor] remove the c implementation of flow_warp

### PR DESCRIPTION
**Pull request by nbei**
Fix bugs in `flow_warp` function in a way of replacing it with numpy implementation.

### CODE REVIEW

**Comment by hellock** [comment]
You may need to check if the documentation needs to be updated accordingly.

File: mmcv/video/optflow.py [hunk 1]
```diff
@@ -151,19 +150,43 @@ def flow_warp(img, flow, filling_value=0, interpolate_mode='nearest'):
...
-    img_float = img.astype(np.float64)
-
-    out = flow_warp_c(
-        img_float,
-        flow.astype(np.float64),
-        filling_value=filling_value,
-        interpolate_mode=interpolate_mode)
-
-    return out
+    height = flow.shape[0]
```
[the remark this answers is not in the document]
Comment (hellock) [reply mmcv/video/optflow.py:153]: and add a warning that this method is only for prototyping

File: setup.py [hunk 2]
```diff
@@ -1,15 +1,11 @@
 import glob
 import os
 import re
-import setuptools
 from pkg_resources import DistributionNotFound, get_distribution
 from setuptools import dist, find_packages, setup
 
 dist.Distribution().fetch_build_eggs(['Cython', 'numpy>=1.11.1'])
```
Comment (hellock) [inline setup.py:7]: Cython is not needed.

File: docs/build.md [hunk 3]
```diff
@@ -101,11 +101,11 @@ MMCV can be built in three ways:
 
 1. Lite version (without ops)
 
-   In this way, only `flow_warp` module will be compiled as a Cython extension.
+   In this way, there does not exist any compilation.
```
Comment (hellock) [inline docs/build.md:104]: In this way, no custom ops are compiled and mmcv is a pure python package.

File: setup.py [hunk 4]
```diff
@@ -1,14 +1,8 @@
...
-from setuptools import dist, find_packages, setup
-
-dist.Distribution().fetch_build_eggs(['Cython', 'numpy>=1.11.1'])
-
-import numpy  # NOQA: E402  # isort:skip
-from Cython.Build import cythonize  # NOQA: E402  # isort:skip
+from setuptools import find_packages, setup
 
 EXT_TYPE = ''
 try:
```
Comment (hellock) [inline setup.py:8]: Since no cython compilation is needed, the following logic can be simplified.

Comment (nbei) [reply setup.py:8]: Since we need to support non-torch compilation, I may think the logic should be kept as the previous one.

Comment (hellock) [reply setup.py:8]: We can just avoid the dependency of cython like the current implementation.

**Review by hellock — APPROVED** [review]

**Review by ycxioooong — APPROVED** [review]

**Pull request merged**